File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/uiperformance/UIPerformanceDisplayDokitView.java
package com.didichuxing.doraemonkit.kit.uiperformance;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.model.ViewInfo;
import com.didichuxing.doraemonkit.kit.core.AbsDokitView;
import com.didichuxing.doraemonkit.kit.core.DokitViewLayoutParams;
import com.didichuxing.doraemonkit.kit.viewcheck.LayoutBorderView;

import java.util.List;

/**
 * Created by jintai on 2019/09/26.
 */
public class UIPerformanceDisplayDokitView extends AbsDokitView implements UIPerformanceManager.PerformanceDataListener {
    private LayoutBorderView mLayoutBorderView;

    @Override
    public View onCreateView(Context context, FrameLayout view) {
        return LayoutInflater.from(view.getContext()).inflate(R.layout.dk_float_ui_performance_display, view, false);
    }

    @Override
    public void onViewCreated(FrameLayout view) {
        mLayoutBorderView = findViewById(R.id.rect_view);
        //设置不响应触摸事件
        setDoKitViewNotResponseTouchEvent(getDoKitView());
    }


    @Override
    public void initDokitViewLayoutParams(DokitViewLayoutParams params) {
        params.flags = DokitViewLayoutParams.FLAG_NOT_FOCUSABLE_AND_NOT_TOUCHABLE;
        params.width = DokitViewLayoutParams.MATCH_PARENT;
        params.height = DokitViewLayoutParams.MATCH_PARENT;
    }

    @Override
    public void onCreate(Context context) {
        UIPerformanceManager.getInstance().addListener(UIPerformanceDisplayDokitView.this);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        UIPerformanceManager.getInstance().removeListener(this);
    }

    @Override
    public void onRefresh(List<ViewInfo> infos) {
        mLayoutBorderView.showViewLayoutBorder(infos);
    }


    @Override
    public boolean canDrag() {
        return false;
    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/uiperformance/UIPerformanceDisplayDokitView.kt
package com.didichuxing.doraemonkit.kit.uiperformance

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.core.AbsDokitView
import com.didichuxing.doraemonkit.kit.core.DokitViewLayoutParams
import com.didichuxing.doraemonkit.kit.uiperformance.UIPerformanceManager.Companion.instance
import com.didichuxing.doraemonkit.kit.uiperformance.UIPerformanceManager.PerformanceDataListener
import com.didichuxing.doraemonkit.kit.viewcheck.LayoutBorderView
import com.didichuxing.doraemonkit.model.ViewInfo

/**
 *
 * Desc:ui界面层级信息相关
 * <p>
 * Date: 2020-06-15
 * Company:
 * Updater:
 * Update Time:
 * Update Comments:
 *
 * Author: pengyushan
 */
class UIPerformanceDisplayDokitView : AbsDokitView(), PerformanceDataListener {
    private var mLayoutBorderView: LayoutBorderView? = null
    override fun onCreateView(context: Context?, view: FrameLayout?): View {
        return LayoutInflater.from(view!!.context).inflate(R.layout.dk_float_ui_performance_display, view, false)
    }

    override fun onViewCreated(view: FrameLayout?) {
        mLayoutBorderView = findViewById(R.id.rect_view)
        //设置不响应触摸事件
        setDokitViewNotResponseTouchEvent(rootView)
    }

    override fun initDokitViewLayoutParams(params: DokitViewLayoutParams?) {
        params!!.flags = DokitViewLayoutParams.FLAG_NOT_FOCUSABLE_AND_NOT_TOUCHABLE
        params.width = DokitViewLayoutParams.MATCH_PARENT
        params.height = DokitViewLayoutParams.MATCH_PARENT
    }

    override fun onCreate(context: Context?) {
        instance.addListener(this@UIPerformanceDisplayDokitView)
    }

    override fun onDestroy() {
        super.onDestroy()
        instance.removeListener(this)
    }

    override fun onRefresh(infos: List<ViewInfo>?) {
        mLayoutBorderView!!.showViewLayoutBorder(infos)
    }

    override fun canDrag(): Boolean {
        return false
    }
}